solr/contrib/ltr/src/java/org/apache/solr/ltr/store/rest/ManagedModelStore.java

```diff
@@ -299,36 +208,95 @@ public String toString() {
+      Map<String,Object> modelMap, ManagedFeatureStore managedFeatureStore) {
+
+    final FeatureStore featureStore =
+        managedFeatureStore.getFeatureStore((String) modelMap.get(STORE_KEY));
+
+    final List<Feature> features = new ArrayList<>();
+    final List<Normalizer> norms = new ArrayList<>();
+
+    final List<Object> featureList = (List<Object>) modelMap.get(FEATURES_KEY);
+    if (featureList != null) {
+      for (final Object feature : featureList) {
+        final Map<String,Object> featureMap = (Map<String,Object>) feature;
+        features.add(featureFromFeatureMap(solrResourceLoader, featureMap, featureStore));
+        norms.add(normalizerFromFeatureMap(solrResourceLoader, featureMap));
       }
-      modelMap.put("features", features);
-      modelMap.put("params", modelmeta.getParams());
+    }
 
-      list.add(modelMap);
+    LTRScoringModel model;
+    try {
+      model = LTRScoringModel.getInstance(solrResourceLoader,
+          (String) modelMap.get(CLASS_KEY), // modelClassName
+          (String) modelMap.get(NAME_KEY), // modelName
+          features,
+          norms,
+          featureStore.getName(),
+          featureStore.getFeatures(),
+          (Map<String,Object>) modelMap.get(PARAMS_KEY));
+    } catch (ModelException ex) {
+      log.info("ModelException {}", ex);
+      throw ex;
     }
-    return list;
+
+    return model;
   }
-  
-  private static Normalizer fromNormalizerMap(SolrResourceLoader solrResourceLoader,
+
+  private static LinkedHashMap<String,Object> toLTRScoringModelMap(LTRScoringModel model) {
+    final LinkedHashMap<String,Object> modelMap = new LinkedHashMap<>(5, 1.0f);
+
+    modelMap.put(NAME_KEY, model.getName());
+    modelMap.put(CLASS_KEY, model.getClass().getCanonicalName());
+    modelMap.put(STORE_KEY, model.getFeatureStoreName());
+
+    final List<Map<String,Object>> features = new ArrayList<>();
+    final List<Feature> featuresList = model.getFeatures();
+    final List<Normalizer> normsList = model.getNorms();
+    for (int ii=0; ii<featuresList.size(); ++ii) {
+      features.add(toFeatureMap(featuresList.get(ii), normsList.get(ii)));
+    }
+    modelMap.put(FEATURES_KEY, features);
+
+    modelMap.put(PARAMS_KEY, model.getParams());
+
+    return modelMap;
+  }
+
+  private static Feature featureFromFeatureMap(SolrResourceLoader solrResourceLoader,
+      Map<String,Object> featureMap, FeatureStore featureStore) {
+    final String featureName = (String)featureMap.get(NAME_KEY);
+    return (featureName == null ? null
+        : featureStore.get(featureName));
+  }
+
+  @SuppressWarnings("unchecked")
+  private static Normalizer normalizerFromFeatureMap(SolrResourceLoader solrResourceLoader,
+      Map<String,Object> featureMap) {
+    final Map<String,Object> normMap = (Map<String,Object>)featureMap.get(NORM_KEY);
+    return  (normMap == null ? IdentityNormalizer.INSTANCE
```

Comment:
I think fromNormalizerMap() should be doing this check, and then you can delete this function and just have the caller of this call fromNormalizerMap() directly.  You have 2 functions that do pretty much the same thing, so I dont see much added value in this function.